### ISSUE TITLE
Disable autorun on startup by default, add app-level configuration

### ISSUE THREAD

**Issue by wasimsandhu** [marimo-team/marimo#2884]
### Description

"Autorun on startup" is a user-level configuration that is enabled by default, and thus poses a security risk for marimo users.

Many (if not all) marimo users are former Jupyter notebook users, and in my experience, the expected behavior of opening a notebook is akin to opening a script file: you do not expect it to run just by viewing it.

With marimo, that is not the case. By default, notebooks run when you open them, which paves the way for attackers to share notebooks that execute malicious code to unsuspecting users.

Even worse is the fact that the cells containing malicious scripts can be "hidden" from the user by being at the bottom of the notebook (out of view when a notebook is first opened), but being executed straight away because it references a variable from one of the first cells run.

### Suggested solution

**Disable autorun on startup by default, and implement app-level autorun configurations.**

By disabling autorun on startup, we protect marimo users from malicious attacks spawning from notebook launches.

By implementing a notebook-level configuration for autorun, we allow marimo users to decide which notebooks they trust to open and execute immediately.

Here's a quick table showing the expected behavior of these configurations:

| User Config | App Config | Result                           |
|-------------|------------|----------------------------------|
| Disabled    | Disabled   | Notebook does NOT run on launch  |
| Enabled     | Disabled   | Notebook does NOT run on launch. |
| Disabled    | Enabled    | Notebook does NOT run on launch. |
| Enabled     | Enabled    | Notebook runs on launch.         |

TLDR: The notebook can only be autorun on startup if BOTH user config and app config permit it.

**Comment by mscolnick** [comment]
Thanks for the explanation. I'm not sure how this helps with security. Seems like the same amount of security as before. Users can just turn off autorun.

**Comment by mscolnick** [comment]
Yea I agree, and something we'd like to solve. That seems sperate so maybe you could open in another PR?

**Comment by leventov** [comment]
https://github.com/marimo-team/marimo/issues/3054 is related because "all cells are cached" has the same (visual) effect as disable auto-run on start-up, especially after https://github.com/marimo-team/marimo/issues/3082 is fixed.

**Comment by leventov** [comment]
Possibly, this issue supercedes #2100?

**Comment by mscolnick** [comment]
@leventov, yea once https://github.com/marimo-team/marimo/issues/3054 and https://github.com/marimo-team/marimo/issues/3082 are done. I think having it off by default would be nice default.

I think after all this is in, we can revisit https://github.com/marimo-team/marimo/issues/2100, but will leave it open for now.

**Comment by uwu-420** [comment]
I'm highly in favor of disabling autorun on startup by default.
Just re-opened a notebook to check what it does and it instantly overwrote some result files I generated a while ago which took a few hours to generate, so I'll have to wait again for a while 😢
Sure, I know better now, but I suspect I'm not the only one who tripped over this.
So yeah, besides security implications, to which I also agree, this is another UX problem to consider.

**Comment by mscolnick** [comment]
This is now done via https://github.com/marimo-team/marimo/pull/6849
